### PR TITLE
[LED] fix global_attention_mask not being passed for generation and docs clarification about grad checkpointing

### PR DESCRIPTION
**Pull request by caesar-one**
# What does this PR do?
There are two changes:
1. The `global_attention_mask` parameter was not passed in `.prepare_inputs_for_generation(...)`. So, when you call the `.generate(...)` method it never reaches the `LEDModel.forward(...)` (even though `global_attention_mask` is passed in `kwargs`). So local attention is used for all tokens. This problem was mentioned also in [this pull request](https://github.com/huggingface/transformers/pull/16485), but not completely fixed.
2. The [documentation of LED](https://huggingface.co/docs/transformers/model_doc/led) states: 
> To fine-tune LED on all 16384, it is necessary to enable gradient checkpointing by executing model.gradient_checkpointing_enable(). 

This could be misleading since gradient checkpointing _can_ be used if training leads to OOM errors, but it is _not actually necessary_, as mentioned in [this issue](https://github.com/huggingface/transformers/issues/16541). Some hints on the use of the `use_cache` flag were also added, as suggested in [this notebook](https://colab.research.google.com/drive/12LjJazBl7Gam0XBPy_y0CTOJZeZ34c2v?usp=sharing) by @patrickvonplaten (who initializes the model using `AutoModelForSeq2SeqLM.from_pretrained("allenai/led-base-16384", gradient_checkpointing=True, use_cache=False)`.

Fixes # (issue)

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten @sgugger

### CODE REVIEW

**Comment by JohnGiorgi** [comment]
Thanks for catching this @caesar-one, looking forward to it getting merged as I am also trying to use LED-based models with a global attention mask.

**Review by patrickvonplaten — APPROVED** [review]
@ydshieh could you also take a quick look? :-)

File: docs/source/en/model_doc/led.mdx [hunk 1]
```diff
@@ -44,8 +44,10 @@ Tips:
 - LED makes use of *global attention* by means of the `global_attention_mask` (see
   [`LongformerModel`]). For summarization, it is advised to put *global attention* only on the first
   `<s>` token. For question answering, it is advised to put *global attention* on all tokens of the question.
-- To fine-tune LED on all 16384, it is necessary to enable *gradient checkpointing* by executing
-  `model.gradient_checkpointing_enable()`.
+- To fine-tune LED on all 16384, *gradient checkpointing* can be enabled in case training leads to out-of-memory (OOM)
+  errors. This can be done by either executing `model.gradient_checkpointing_enable()` or using the 
+  `gradient_checkpointing=True` flag in `.from_pretrained(...)` when loading the model. Moreover, the `use_cache=False`
```
Comment (ydshieh) [inline docs/source/en/model_doc/led.mdx:49]: I am not very sure about 
```
`gradient_checkpointing=True` flag in `.from_pretrained(...)` when loading the model.
```
as I see
https://github.com/huggingface/transformers/blob/b0733457a901e5ac197a8e8f183ea25d972801cd/src/transformers/configuration_utils.py#L361-L367

It looks like we would rather use `gradient_checkpointing` for `TrainingArguments`.

**Review by ydshieh — APPROVED** [review]
LGTM, but left a question for the documentation. @patrickvonplaten Could you double check?

Thanks for the fix in this PR, @caesar-one !

**Comment by caesar-one** [comment]
@patrickvonplaten could this issue be the reason for the [lower performances of Primera](https://huggingface.co/allenai/PRIMERA) (LED-based model) in the hf implementation vs the original implementation? After all, global attention was not considered during generation.

Anyway, thanks for your fast feedback :-)

File: docs/source/en/model_doc/led.mdx (hunk 1 above)
Comment (patrickvonplaten) [inline docs/source/en/model_doc/led.mdx:49]:
```suggestion
. Moreover, the `use_cache=False`
```

File: docs/source/en/model_doc/led.mdx [hunk 2]
```diff
@@ -44,8 +44,10 @@ Tips:
 - LED makes use of *global attention* by means of the `global_attention_mask` (see
   [`LongformerModel`]). For summarization, it is advised to put *global attention* only on the first
   `<s>` token. For question answering, it is advised to put *global attention* on all tokens of the question.
-- To fine-tune LED on all 16384, it is necessary to enable *gradient checkpointing* by executing
-  `model.gradient_checkpointing_enable()`.
+- To fine-tune LED on all 16384, *gradient checkpointing* can be enabled in case training leads to out-of-memory (OOM)
+  errors. This can be done by either executing `model.gradient_checkpointing_enable()` or using the
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/led.mdx:48]:
```suggestion
  errors. This can be done by either executing `model.gradient_checkpointing_enable()`. 
```

File: docs/source/en/model_doc/led.mdx (hunk 1 above)
Comment (patrickvonplaten) [reply docs/source/en/model_doc/led.mdx:49]: Agree with @ydshieh, it should be passed to the `TrainingArguments`

File: docs/source/en/model_doc/led.mdx [hunk 3]
```diff
@@ -44,8 +44,10 @@ Tips:
 - LED makes use of *global attention* by means of the `global_attention_mask` (see
   [`LongformerModel`]). For summarization, it is advised to put *global attention* only on the first
   `<s>` token. For question answering, it is advised to put *global attention* on all tokens of the question.
-- To fine-tune LED on all 16384, it is necessary to enable *gradient checkpointing* by executing
-  `model.gradient_checkpointing_enable()`.
+- To fine-tune LED on all 16384, *gradient checkpointing* can be enabled in case training leads to out-of-memory (OOM)
+  errors. This can be done by either executing `model.gradient_checkpointing_enable()`.
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/led.mdx:48]:
```suggestion
  errors. This can be done by executing `model.gradient_checkpointing_enable()`. 
```

File: docs/source/en/model_doc/led.mdx [hunk 4]
```diff
@@ -44,8 +44,10 @@ Tips:
 - LED makes use of *global attention* by means of the `global_attention_mask` (see
   [`LongformerModel`]). For summarization, it is advised to put *global attention* only on the first
   `<s>` token. For question answering, it is advised to put *global attention* on all tokens of the question.
-- To fine-tune LED on all 16384, it is necessary to enable *gradient checkpointing* by executing
-  `model.gradient_checkpointing_enable()`.
+- To fine-tune LED on all 16384, *gradient checkpointing* can be enabled in case training leads to out-of-memory (OOM)
+  errors. This can be done by either executing `model.gradient_checkpointing_enable()`. 
+. Moreover, the `use_cache=False`
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/led.mdx:49]:
```suggestion
 Moreover, the `use_cache=False`
```

**Comment by patrickvonplaten** [comment]
Thanks a lot for the PR @caesar-one !
